src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -43,7 +43,14 @@ public sealed class NodeHandler : Handler, INodeHandler
             string workingDirectory = Data.WorkingDirectory;
             if (string.IsNullOrEmpty(workingDirectory))
             {
-                workingDirectory = TaskDirectory;
+                if (!string.IsNullOrEmpty(ExecutionContext.Variables.System_DefaultWorkingDirectory))
+                {
```

Comment:
OK.  I still don't like TaskDirectory.  Is there another variable like the agent's root working directory?